lib/remote.provider.js

```diff
@@ -1,7 +1,148 @@
+
+const APP_NAME = "GLOBAL_STORAGE";
+const INTERFACES = ["provider", "cursor"];
+
+class RemoteProvider extends StorageProvider {
+  // Create RemoteProvider
+  // cursor - RemoteCursor instance
+  // apiPromise - jstp promise
+  constructor() {
+    super();
+    this.api = null;
+    this.cursors = [];
+  }
+
+  // Open StorageProvider
+  //   callback - function(err, RemoteProvider)
+  // eslint-disable-next-line no-unused-vars
+  open(options, callback) {
+    jstp.ws.connectAndInspect(
+      APP_NAME,
+      null,
+      INTERFACES,
+      options,
+      (error, connection, api) => {
+        if (error) {
+          callback(error);
+          return;
+        } else {
+          this.api = api;
+          callback(null, this);
+        }
+      }
+    );
+  }
+
+  // Close RemoteProvider
+  //   callback - function(err)
+  // eslint-disable-next-line no-unused-vars
+  close(callback) {
+    if (!this.api) {
+      callback();
+      return;
+    }
+    this.api = null;
+  }
+
+  // Generate globally unique id
+  //   callback - function(err, id)
+  // eslint-disable-next-line no-unused-vars
+  takeId(callback) {
+    throw new GSError(errorCodes.NOT_IMPLEMENTED);
+  }
+
+  // Get record from GlobalStorage
+  //   id - globally unique record id
+  //   callback - function(err, obj)
+  // eslint-disable-next-line no-unused-vars
+  get(id, callback) {
+    this.api.provider.get(id, (error, callback));
+
+    getRecordMethod(id).then(record => callback(record));
+  }
+
+  // Set record in GlobalStorage
+  //   record - record to be stored
+  //   callback - function(err)
+  // eslint-disable-next-line no-unused-vars
+  set(record, callback) {
+    const setRecordMethod = createMethod(
+      api => api.provider.set,
+      this.apiPromise
+    );
+
+    setRecordMethod(record).then(() => callback());
+  }
+
+  // Create record in GlobalStorage
+  //   record - record to be stored
+  //   callback - function(err, id)
+  // eslint-disable-next-line no-unused-vars
+  create(category, record, callback) {
+    this.api.provider.create(
+      category,
+      record,
+      (error,
+      id => {
+        if (error) {
+          callback(error);
+          return;
+        }
+        callback(null, id);
+      })
+    );
+  }
+
+  // Update record in GlobalStorage
+  //   query - record, example: { Id }
+  //   patch - record, fields to update
+  //   callback - function(err, count)
+  // eslint-disable-next-line no-unused-vars
+  update(category, query, patch, callback) {
+    this.api.provider.update(category, query, patch, (error, count) => {
+      if (error) {
+        callback(error);
+        return;
+      }
+      callback(null, count);
+    });
+  }
+
+  // Delete record in GlobalStorage
+  //   query - record, example: { Id }
+  //   callback - function(err, count)
+  // eslint-disable-next-line no-unused-vars
+  delete(category, query, callback) {
+    this.api.provider.delete(category, query, (error, count) => {
```

Comment:
You can pass callback straight ahead, there is no need to create a new function.